lib/daru/category.rb

```diff
@@ -918,10 +918,27 @@ def modify_category_at pos, category
       end
       old_category = category_from_position pos
       @array[pos] = int_from_cat category
-      @cat_hash[old_category].delete pos
+      @cat_hash[old_category].delete pos unless old_category.nil?
       @cat_hash[category] << pos
     end
 
+    def set pos, category
+      unless categories.include? category
+        raise ArgumentError, "Invalid category #{category}, "\
+          'to add a new category use #add_category'
+      end
+      pos = pos.is_a?(Array) ? pos : [pos]
+      pos.each do |x|
+        if @index.include? x
+          self.[]= x, category
+          next
+        end
+        @array << categories.index(category)
+        @index = @index.add(x)
```

Comment:
I think it will be better to use cat_to_int function here

Reply:
I think you mean `int_to_cat`. Please review the changes.